### PR TITLE
RuboCop 1.74 Unsafe Correctables 3/3 

### PR DESCRIPTION
**Pull request by JoeCohen**
Fixes unsafe correctable offenses. Part 3 of 3. 
Broken into 3 parts for ease of review. This part includes cops linked below. 
For each putative offense:
- Accepts the fix if safe, or disables the cop if unsafe
- Makes follow-on changes to avoid new offenses. (Ex: removes `.to_s` inside string interpolation, divides long lines.)
- Updates .rubocop_todo

#### Cops in this PR
Check the linked documentation for comments about cop safety.
> [Rails/Blank](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsblank)
[Rails/Output](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsoutput)
[Rails/RakeEnvironment](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsrakeenvironment)
[Rails/RedundantActiveRecordAllMethod](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsredundantactiverecordallmethod)
[Rails/ResponseParsedBody](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsresponseparsedbody)
[Rails/RootPathnameMethods](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsrootpathnamemethods)
[Rails/SelectMap](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railsselectmap)
[Style/BitwisePredicate](https://docs.rubocop.org/rubocop/1.74/cops_style.html#stylebitwisepredicate)
[Style/CombinableLoops](https://docs.rubocop.org/rubocop/1.74/cops_style.html#stylecombinableloops)
[Style/FileNull](https://docs.rubocop.org/rubocop/1.74/cops_style.html#stylefilenull)[Style/SlicingWithRange](https://docs.rubocop.org/rubocop/1.74/cops_style.html#styleslicingwithrange)
[Style/HashEachMethods](https://docs.rubocop.org/rubocop/1.74/cops_style.html#stylehasheachmethods)
[Style/SafeNavigation](https://docs.rubocop.org/rubocop/1.74/cops_style.html#stylesafenavigation)

### CODE REVIEW

File: app/extensions/string.rb [hunk 1]
```diff
@@ -739,10 +739,6 @@ def md5sum
 
   ### Misc Utilities ###
   #
-  def print_thing(thing)
```
Comment (JoeCohen) [inline app/extensions/string.rb:742]: `print_thing` isn't called anywhere.

File: app/models/name/spelling.rb [hunk 2]
```diff
@@ -35,7 +35,7 @@ def suggest_alternate_spellings(str)
         results = guess_word("", words.first)
         (2..num).each do |i|
           next unless results.any?
-          next unless (i & 1).zero?
+          next unless i.even?
```
Comment (JoeCohen) [inline app/models/name/spelling.rb:38]: @nimmolo : Is my change okay?
RuboCop wanted me to change `(i & 1).zero?` to `i.nobits?(1)`. https://docs.rubocop.org/rubocop/1.74/cops_style.html#stylebitwisepredicate
But both seem like extra mental gymnastics.

File: config/initializers/monkey_patches.rb [hunk 3]
```diff
@@ -6,7 +6,7 @@
 require("action_view/log_subscriber")
 
 # Require all Ruby files in the core_extensions directory
-Dir[Rails.root.join("lib/core_extensions/**/*.rb")].each { |f| require f }
+Rails.root.glob("lib/core_extensions/**/*.rb").each { |f| require f }
```
Comment (JoeCohen) [inline config/initializers/monkey_patches.rb:9]: @nimmolo Please double-check

File: lib/tasks/lang.rake [hunk 4]
```diff
@@ -78,7 +78,7 @@ namespace :lang do
   task setup: [:environment, :login, :verbose, :safe_mode]
 
   desc "Log in user for import tasks."
-  task(:login) do
+  task(login: :environment) do
```
Comment (JoeCohen) [inline lib/tasks/lang.rake:81]: @mo-nathan @pellaea 
I added `:environment` as a dependency to 4 tasks.
- Question: Are these tasks ever used? 

- I don't understand the tasks.
But without this dependency, I got
```ruby
joe@iMac mushroom-observer % rake lang:login_admin
rake aborted!
NameError: uninitialized constant User (NameError)

    User.current = User.find(0)
    ^^^^
/Users/joe/mushroom-observer/lib/tasks/lang.rake:91:in `block (2 levels) in <top (required)>'
Tasks: TOP => lang:login_admin
(See full trace by running task with --trace)
```

**Comment by JoeCohen** [comment]
@mo-nathan & @nimmolo:
I tagged you in comments where I had special questions.

File: lib/tasks/lang.rake (hunk 4 above)
Comment (mo-nathan) [reply lib/tasks/lang.rake:81]: I for sure don't intentionally use them.  However, they might get called indirectly.  In my view our translation stuff is kinda of wonky so I would move carefully.

File: app/models/name/spelling.rb (hunk 2 above)
Comment (nimmolo) [reply app/models/name/spelling.rb:38]: Thanks for asking. I don't understand what this is doing, though.

File: app/extensions/string.rb (hunk 1 above)
Comment (nimmolo) [reply app/extensions/string.rb:742]: Seems useful (and intended) for debugging, so it wouldn't stay in the code base. But i didn't know about it.

File: app/models/name/spelling.rb (hunk 2 above)
Comment (mo-nathan) [reply app/models/name/spelling.rb:38]: It's doing a bitwise and of i against the number 1 and then testing if that equals zero.  This will be the case if and only if i is even.  Why it was written the way it is I have no idea.  `git blame` current points @nimmolo , but I'm guessing it was copied from somewhere else.

File: app/models/image.rb [hunk 5]
```diff
@@ -662,7 +662,10 @@ def validate_image_name
   # to the :image field.  Returns true if the file is successfully saved.
   def save_to_temp_file
     result = true
-    unless upload_temp_file.present?
+    # disable cop; In theory upload_temp_file could be
+    # a blank non-empty string (e.g. " ").
+    # Don't fix unless...present?; it isn't broken.
+    unless upload_temp_file.present? # rubocop:disable Rails/Blank
```
Comment (mo-nathan) [inline app/models/image.rb:668]: `present?` is literally defined as `!blank?` (see https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/object/blank.rb).  I would go with Rubocop's recommendation here.  For " " `blank?` is true and `present?` is false.  If you want to allow filenames that include `" "` you would have to explicitly test for that.  However, that seems so edge casey that I think it's a bad idea to consider in our code.  If you know this is actually a likely name, then I'd reject it or rename it to something reasonable.

**Review by mo-nathan — APPROVED** [review]
Made a few minor points, but overall this looks great!
